src/main/webapp/resources/script.js

```diff
@@ -12,3 +12,42 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+$(document).ready(function () {
+  var step = 1;
+  var ingredient = 1;
+  addRemove(
+    step,
+    ".step-add-button",
+    ".step-wrapper",
+    ".step-form",
+    ".step-remove-button"
+  );
+  addRemove(
+    ingredient,
+    ".ingredient-add-button",
+    ".ingredient-wrapper",
+    ".ingredient-form",
+    ".ingredient-remove-button"
+  );
+});
+
+function addRemove(x, addButton, wrapper, form, removeButton) {
+  const MAX_NUMBER_OF_FIELD = 50;
+  //Once add button is clicked
+  $(addButton).click(function () {
+    var inputBox = $(form).clone();
+    var fieldHTML = "<div>" + inputBox.html() + "</div>"; //New input field html
+    //Check maximum number of input fields
+    if (x < MAX_NUMBER_OF_FIELD) {
+      x++; //Increment field counter
+      $(wrapper).append(fieldHTML); //Add field html
+    }
+  });
+
+  //Once remove button is clicked
+  $(wrapper).on("click", removeButton, function (e) {
+    e.preventDefault();
+    $(this).parent("div").remove(); //Remove field html
+    x--; //Decrement field counter
```

Comment:
Uh, this is really shady, you're preserving state by changing the parameter value.

I'd create a class called "FormItemHandler" that you can instantiate and have the "x" (which you can call numberItems to be more readable).

Other option would be to just query with jquery for the elements and get the number of them in the DOM.

I'd leave to you what to do.

NOTE: x, y, z variables can be only used for very specific cases, example, functions performing mathematical operations. If you're defining a variable with this name that should make you think a better name for it already.

Reply:
Can you clarify this?

Reply:
I have changed x to numberOfForm